OpenEdX/OpenEdX.entitlements

```diff
@@ -8,5 +8,12 @@
 	<array>
 		<string>Default</string>
 	</array>
+	<key>com.apple.developer.associated-domains</key>
+	<array>
+		<string>applinks:edx.app.link</string>
```

Comment:
We/community can add this capability to the whitelabel script to modify the entitlements if needed. But I think we can use edx as a placeholder in the codebase. Thoughts?

Reply:
Please let's move it to the config file if possible. 
It will be much clearer for the community to configure everything from the configuration file.

Reply:
@volodymyr-chekyrta @rnr I've also made the `Associated Domains` configurable and it will be added when Branch is enabled and will also configure values as per `DEEPLINK_PREFIX`

Reply:
@varshamenon4 @rnr I've added the default values as well. The scene of info.plist and entitlements is different. The `info.plist` is baked into the IPA file so the updated file is directly modified inside the IPA, hence to trackable change is there. 

But in the case of entitlements, the changes are directly applied to the entitlements file and they are git trackable if there will be any change because of prefix. That's why I've added `edx` as a placeholder value so that for edX and openedX the change will not appear in the changeset. Otherwise, we have to remove the changes of this file before committing any changes and we can't even add it to git ignore because in the future more entitlements can be added.